### PR TITLE
Edit messages

### PR DESCRIPTION
**Pull request by nbaldy**
Give initial ability to edit messages. Starts a bug in that editing drops files

### CODE REVIEW

**Review by ArgaAlan — COMMENTED** [review]
LGTM Just I would really like if somebody else takes a look because there is a lot of stuff and I might not see something that could cause a problem

**Comment by nbaldy** [comment]
Pre-merged with "fixCountryBreak" for testing, with the thought that fixCountryBreak will be merged into Master first. Until then, however, there will look like there are some extra changes

File: src/main/java/com/google/codeu/data/Datastore.java
Comment (nbaldy) [inline src/main/java/com/google/codeu/data/Datastore.java:66]: These were changed 1 - for consistancy and 2- because they were referenced this way below, which returned null as-it-was

File: src/main/java/com/google/codeu/data/Message.java
Comment (nbaldy) [inline src/main/java/com/google/codeu/data/Message.java:124]: Previous version threw up errors on messages which had been created before the image functionality. This seemed to fix it

**Review by ArgaAlan — APPROVED** [review]
Might be a good idea to try to not merge this PR and mine at the same time because may cause some trouble. Any way LGTM

**Review by drewbern — APPROVED** [review]
LGTM!

**Pull request merged**
